client/electron/go_vpn_tunnel.ts

```diff
@@ -151,32 +161,47 @@ export class GoVpnTunnel implements VpnTunnel {
     }
 
     console.log('restarting tun2socks after resume');
-    await Promise.all([
-      this.tun2socks.start(this.transportConfig, this.isUdpEnabled),
-      this.updateUdpSupport(), // Check if UDP support has changed; if so, silently restart.
-    ]);
+    await this.updateUdpAndRestartTun2socks();
   }
 
-  private async updateUdpSupport() {
-    const wasUdpEnabled = this.isUdpEnabled;
-    try {
-      this.isUdpEnabled = await checkUDPConnectivity(
+  private startTun2socks(): Promise<void> {
+    if (IS_WINDOWS) {
+      return this.tun2socks.startWindows(
         this.transportConfig,
-        this.isDebugMode
+        this.isUdpEnabled,
+        this.gatewayAdapterIndex
```

Comment:
Should gatewayAdapterIndex be in this.tun2socks instead so you can just call start()?

Reply:
I'm trying to align with other parameters (like `isUdpSupported`), I think keeping them in the same place would be cleaner.